### DISCUSSION TITLE
Deploy Express js

### DISCUSSION THREAD

**Discussion by arturyushchenko** [vercel/community#2834] [Help]
### Summary

hello, we are trying to raise an express js project, but nothing happens, apparently something with the project building, I am attaching the files below.

WARN! Due to builds existing in your configuration file, the Build and Development Settings defined in your Project Settings will not apply. Learn More: https://vercel.link/unused-build-settings

I've tried all the solutions on the internet and nothing worked.

### Example

workella-back-end.vercel.app

### Steps to Reproduce

hello, we are trying to raise an express js project, but nothing happens, apparently something with the project building, I am attaching the files below.

WARN! Due to builds existing in your configuration file, the Build and Development Settings defined in your Project Settings will not apply. Learn More: https://vercel.link/unused-build-settings

fiile index.js

```javascript
require('dotenv').config();

const express = require('express');
const sequelize = require('./src/db');
const cors = require('cors');
const cron = require('node-cron');
const fileUpload = require("express-fileupload")

const router = require('./src/routes/index');
const sseService = require('./src/services/sseEvents/SseService');
const errorHandler = require('./src/middleware/errorMiddleware');

const PORT = process.env.PORT || 5000;
const app = express();

app.use(cors({
    exposedHeaders: ['Content-Disposition'],
}));

app.use(fileUpload({
    createParentPath: true
}))
app.use(express.static('src/uploads'));
app.use(express.json());
app.use('/api', router);

app.use(errorHandler);

const start = async () => {
    try {
        await sequelize.authenticate();
        await sequelize.sync();
        app.listen(PORT, () => console.log(`Server started on port ${PORT}`));

        //SSE ping clients
        cron.schedule('*/15 * * * * *', sseService.pingClient.bind(sseService));
    } catch (error) {
        console.log(error);
    }
}

start();
``` 

file package.json

```javascript
{
  "name": "server",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "engines": {
    "node": "18.x"
  },
  "scripts": {
    "start": "node index.js",
    "dev": "nodemone index.js"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "aws-sdk": "^2.1363.0",
    "axios": "^1.4.0",
    "bcrypt": "^5.1.0",
    "cookie-parser": "^1.4.6",
    "cors": "^2.8.5",
    "dotenv": "^16.0.3",
    "express": "^4.18.2",
    "express-fileupload": "^1.4.0",
    "google-auth-library": "^8.7.0",
    "google-cal-meet-api": "^1.1.7",
    "googleapis": "^114.0.0",
    "handlebars": "^4.7.7",
    "joi": "^17.7.0",
    "jsonwebtoken": "^9.0.0",
    "jszip": "^3.10.1",
    "madge": "^6.0.0",
    "moment-timezone": "^0.5.43",
    "node-cron": "^3.0.2",
    "nodemailer": "^6.9.1",
    "pg": "^8.8.0",
    "pg-hstore": "^2.3.4",
    "sequelize": "^6.28.0",
    "uuid": "^9.0.0"
  },
  "devDependencies": {
    "nodemon": "^2.0.20"
  }
}

``` 

file vercel.json
```javascript
{
  "version": 2,
  "builds": [
    {
      "src": "index.js",
      "use": "@vercel/node"
    }
  ],
  "routes": [
    {
      "src": "/(.*)",
      "dest": "/index.js"
    }
  ]
}
``` 

I've tried all the solutions on the internet and nothing worked.

**Comment by smaeda-ks** [comment]
@Arthurysh have you moved the express entrypoint file (index.js) into the `/api` folder? That's important.
https://vercel.com/guides/using-express-with-vercel#standalone-express

Otherwise, please create a minimal reproducible example and share that with us:
https://vercel.com/guides/creating-a-minimal-reproducible-example

**Reply by arturyushchenko** [reply]
here are screenshots of the structure
![telegram-cloud-photo-size-2-5208909802737092331-x](https://github.com/vercel/community/assets/61291332/da1c71ba-a069-4eb2-a76d-239fc3471fb5)
![telegram-cloud-photo-size-2-5208909802737092330-y](https://github.com/vercel/community/assets/61291332/8d56717f-433c-4791-9ef5-e8e803f430a4)

**Comment by MeetBhingradiya** [comment]
@Arthurysh  If You have Repository connected with your Vercel project Frist Disconnect it because this settings not work with Git Repositorys for now !

use Vercel Cli Instead of Using GitHub Repository Connection ! 

Your Settings is Good If you use Express as API Only Then Below Settings For You.

```
{
    "version": 2,
    "builds": [
        {
            "src": "index.js",
            "use": "@vercel/node"
        }
    ],
    "rewrites": [
        {
            "source": "(.*)",
            "destination": "index.js"
        }
    ],
    "headers": [
        {
            "source": "(.*)",
            "headers": [
                {
                    "key": "Access-Control-Allow-Credentials",
                    "value": "true"
                },
                {
                    "key": "Access-Control-Allow-Origin",
                    "value": "*"
                },
                {
                    "key": "Access-Control-Allow-Methods",
                    "value": "GET,OPTIONS,PATCH,DELETE,POST,PUT"
                },
                {
                    "key": "Access-Control-Allow-Headers",
                    "value": "X-CSRF-Token, X-Requested-With, Accept, Accept-Version, Content-Length, Content-MD5, Content-Type, Date, X-Api-Version"
                }
            ]
        }
    ]
}
```
